dnf-docker-test/features/steps/file_steps.py

```diff
@@ -135,3 +137,33 @@ def step_an_ini_file_filepath_modified_with(ctx, filepath):
                 else:
                     conf.set(section, key, settings[key])
     file_utils.create_file_with_contents(filepath, conf)
+
+@given('I run steps from file "{filepath}"')
+def step_i_run_steps_from_file(ctx, filepath):
+    """
+    Runs all the steps from the given file (``{filepath}``).
+    The file with steps is searched at a provided location
+    and in case it is not found, the filename is searched also
+    in a directory saved in the BEHAVE_FEATUREDIR environment
+    variable or under the /tests directory.
+    *.feature files but should not have *.feature suffix
+    so it won't be executed on its own. Also, the file
+    should contain only steps, there should be no
+    Feature: or Scenario: descriptions.
+    """
+    if not os.path.isfile(filepath):  # there is no such file, I will try to check other locations
+        # first check if the user exported the environment variable BEHAVE_FEATUREDIR
+        filename = os.path.basename(filepath)
+        if os.environ.get('BEHAVE_FEATUREDIR') and os.path.isfile('%s/%s' % (os.environ.get('BEHAVE_FEATUREDIR'), filename)):
+            filepath = '%s/%s' % (os.environ.get('BEHAVE_FEATUREDIR'), filename)
+        # try the /tests/ directory used in Docker image
```

Comment:
honestly, I don't really like this

Reply:
if BEHAVE_FEATUREDIR is set and doesn't exist, fail (raise an exception)
consider using os.path.join(..., filename.lstrip("/")) instead of "%s/%s"

Reply:
I will update the code to use os.path.join, however os.path.isfile doesn't raise an exception and therefore I do not understand your comment.